src/System.Private.Xml.Linq/src/System/Xml/Linq/XHashtable.cs

```diff
@@ -397,22 +398,9 @@ private bool FindEntry(int hashCode, string key, int index, int count, ref int e
             /// </summary>
             private static int ComputeHashCode(string key, int index, int count)
             {
-                int hashCode = StartingHash;
-                int end = index + count;
                 Debug.Assert(key != null, "key should have been checked previously for null");
-
-                // Hash the key
-                for (int i = index; i < end; i++)
-                    unchecked
-                    {
-                        hashCode += (hashCode << 7) ^ key[i];
-                    }
-
-                // Mix up hash code a bit more and clear the sign bit.  This code was taken from NameTable.cs in System.Xml.
-                hashCode -= hashCode >> 17;
-                hashCode -= hashCode >> 11;
-                hashCode -= hashCode >> 5;
-                return hashCode & 0x7FFFFFFF;
+                ReadOnlySpan<byte> bytes = MemoryMarshal.AsBytes(key.AsSpan(index, index + count));
```

Comment:
```cs
return string.GetHashCode(key.AsSpan(index, count));
```

Reply:
I didn't know we added this :) perhaps it is possible to remove Marvin from Common now?

Reply:
This API was added in netcoreapp 3.0. So for things which are inbox or build as _netcoreapp_ or _uapaot_, this API should be accessible. I believe this particular project qualifies.

For things which are fully OOB and which are cross-compiled against _netstandard_ / _netfx_ / etc., the private implementation will have to be carried. So unfortunately we can't delete the private implementation just yet.